catlover.client/src/components/cats/catsComponent.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from "react";
+import CatComponent from "./catComponent";
+import LoadingComponent from "../loading/loadingComponent";
+import ErrorComponent from "../error/errorComponent";
+import PageTitleComponent from "../pageTitle/pageTitleComponent";
+import PaginationComponent from "../pagination/paginationComponent";
+
+class CatsComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentPage: 1,
+      catsPerPage: 10,
+      error: null,
+      isLoaded: false,
+      message: String,
+      items: [],
+    };
+  }
+
+  componentDidMount() {
+    fetch("http://localhost:10000/cats")
+      .then((res) => res.json())
+      .then(
+        (result) => {
+          this.setState({
+            isLoaded: true,
+            message: result.message,
+            items: result.data,
+          });
+        },
+        (error) => {
+          this.setState({
+            isLoaded: true,
+            error,
+          });
+        }
+      );
+  }
+
+  render() {
+    const { currentPage, catsPerPage, error, isLoaded, message, items } =
+      this.state;
+    const indexOfLastCat = currentPage * catsPerPage;
+    const indexOfFirstCat = indexOfLastCat - catsPerPage;
+    const currentCats = items.slice(indexOfFirstCat, indexOfLastCat);
+    const paginate = (pageNum) => this.setState({ currentPage: pageNum });
+    const nextPage = () => this.setState({ currentPage: currentPage + 1 });
+    const prevPage = () => this.setState({ currentPage: currentPage - 1 });
```

Comment:
Do you think all of those declarations could have gone to classes scope? Would there be any problems?

Reply:
I was going to ask the same thing here! Fe experience was really nice except this part. Checking the total page num.

Reply:
Yes, this is better approach because we can create a PaginationClass and then we can use it in every list in the application. This part of code was copy from the net, so i just paste it here and it worked, so i didn't check it again in order to optimize it. My mistake.

Reply:
Correct but I was referring more to the function in which you decided to put this logic. IMHO it would be better to add to class scope instead of function scope